executor/src/witgen/jit/symbolic_expression.rs

```diff
@@ -0,0 +1,319 @@
+use std::{
+    fmt::{self, Display, Formatter},
+    ops::{Add, BitAnd, BitOr, Mul, Neg},
+    rc::Rc,
+};
+
+use powdr_number::FieldElement;
+
+use crate::witgen::range_constraints::RangeConstraint;
+
+/// A value that is known at run-time, defined through a complex expression
+/// involving known cells or variables and compile-time constants.
+/// Each of the sub-expressions can have its own range constraint.
+#[derive(Debug, Clone)]
+pub enum SymbolicExpression<T: FieldElement, V> {
+    /// A concrete constant value known at compile time.
+    Concrete(T),
+    /// A symbolic value known at run-time, referencing either a cell or a local variable.
```

Comment:
What's a "local variable"?

In my current understanding, this could be:
- A cell
- A scalar public (kind of a global variable)
- A challenge (similar to scalar public, but known from the start)

Would "local variable" by an intermediate column?

Reply:
I think I took this comment from the original pull request. There, I introduce local variables in the rust code to implement lookups, because it as more convenient than using the cells right away.

I mean this is a very generic data structure and variable is just a symbolic reference to something. We could also use `Symbol` instead of `Variable`, if you thin that captures it better that it might be known or might be unknown.

Reply:
Makes sense that the code here is more generic and does not need to mention stuff like challenges. And yes, I think `Symbol` would be better!